src/v2/view-builder/views/consent/EmailMagicLinkOTPTerminalView.js

```diff
@@ -0,0 +1,59 @@
+import { loc, View } from 'okta';
+import hbs from 'handlebars-inline-precompile';
+
+const getTerminalOtpEmailMagicLinkContext = (appState) => {
+  const appName = loc('idx.return.link.otponly.accessing.app', 'login', [appState.get('app').label]);
+  const client = appState.get('client');
+  const browserOnOsString = loc('idx.return.link.otponly.browser.on.os', 'login', [client.browser, client.os]);
+  // need city and 1 of (state, country) to display location
+  const clientLocationExists = client.location &&
+   client.location.city && 
+   (client.location.state || client.location.country);
+  const geolocation = clientLocationExists ?
+    loc('idx.return.link.otponly.city.state.country', 
+      'login',
+      [client.location.city, client.location.state || client.location.country]) :
+    null;
+  const otp = appState.get('currentAuthenticator').contextualData.otp;
+
+  return {
+    appName,
+    browserOnOsString,
+    geolocation,
+    otp,
+  };
+};
+
+const BaseEmailMagicLinkOTPTerminalView = View.extend({
+  getTemplateData() {
+    return getTerminalOtpEmailMagicLinkContext(this.options.appState);
+  },
+});
+
+const OTPInformationTerminalView = BaseEmailMagicLinkOTPTerminalView.extend({
+  template: hbs`
+  <h1 class='otp-value'>{{otp}}</h1>
+  <div class="enduser-email-consent--info no-translate">
+    <div>{{i18n code="idx.return.link.otponly.request" bundle="login"}}</div>
+  </div>
+  <div class="enduser-email-consent--info no-translate">
+    <i class="enduser-email-consent--icon icon--desktop"></i>
+    <div>{{browserOnOsString}}</div>
+  </div>
+  <div class="enduser-email-consent--info no-translate">
+    <i class="enduser-email-consent--icon icon--app"></i>
+    <div>{{appName}}</div>
+  </div>
+  {{#if geolocation}}
+  <div class="enduser-email-consent--info no-translate">
+    <i class="enduser-email-consent--icon icon--location"></i>
+    <div>{{geolocation}}</div>
+  </div>
+  {{/if}}
+  <p class='otp-warning'>{{i18n code="idx.return.link.otponly.warning" bundle="login"}}</p>
+  `,
+});
```

Comment:
nit: You shouldn't need both `BaseEmailMagicLinkOTPTerminalView` and `OTPInformationTerminalView`, so I'd suggest combining them:

```suggestion
  `,
  getTemplateData() {
    return getTerminalOtpEmailMagicLinkContext(this.options.appState);
  },
});
```

Reply:
I think in the future we're considering having multiple OTP terminal views which is why I used this approach, should I still merge it for now though?

Reply:
Ah - didn't know this was an optimization for the future. Feel free to keep it!